### ISSUE TITLE
Face / License plate recognition is erratic

### ISSUE THREAD

**Issue by alex9099** [openstreetcam/openstreetview.org#162]
Hi, i was viewing my processed track but the Face / License plate recognition is awful, lot's of thing that are not faces or license plates are blurred, can't the images be reviewed by it's owner? Also can i download the whole track without the blur(assuming that i'm the owner of the photos)?

**Comment by mds08011** [comment]
I agree.  It would be beneficial to have a method for identifying images with incorrect blurring.

For example, I have a photo where a bike route sign was blurred during the image processing. (Sign on right side between the gap in parked cars): http://openstreetcam.org/details/1263209/55/track-info

**Comment by alexandrui-telenav** [comment]
We're working on a machine learning approach ( the one on the market is an open source algorithmic solution ) that will improve this and add support for user reviews. 
It's going to continue to be awful for a while unfortunately..

**Comment by alex9099** [comment]
Also manual tagging would be nice :)

**Comment by jsniff** [comment]
@alexandrui-telenav  @mds08011   
Our company helps other companies with their privacy protection needs. We have rly good face and license blurring software. Would either of you guys be interested in seeing a demo? We built it for the body cam industry originally and now looking for new applications. Would be interesting to see if we can help you guys.

**Comment by mds08011** [comment]
@alexandrui-telenav In the meantime, I think it would be beneficial to allow the sign detection to override blurring.  I have run into some instances where the Sign Detection accurately identifies a sign, but is then blurred over, making it unhelpful for mapping purposes.  This seems to mostly happen with highway signs.

Example 1: http://openstreetcam.org/details/1289493/20/detections
Example 2: http://openstreetcam.org/details/1289493/44/detections
Example 3: http://openstreetcam.org/details/1287947/3/detections

**Comment by alexandrui-telenav** [comment]
Fully agree @mds08011 . The Machine Learning component we developed shows promising test results ( not yet integrated in production ) and potentially allow what you're suggesting. Current architecture is not flexible enough for that.

**Comment by mds08011** [comment]
That makes sense.  Thanks for the update!

**Comment by jsniff** [comment]
@alexandrui-telenav @mds08011

Our company helps other companies with their privacy protection needs. We have rly good face and license blurring software. We've done a lot with body cam companies: http://www.suspecttech.com/ Would either of you guys be interested in seeing a demo next week? What works?

**Comment by mds08011** [comment]
@jsniff I am not associated with the development team.  Given that a solution is already being worked on internally, I am guessing they are not interested in outsourcing this work.

**Comment by jsniff** [comment]
@mds08011 
Malcolm understood. The thing is, having seen so many projects done by large companies, they almost never seem to end well, unless it's their core expertise. Ultimately we're hoping our clients save money by spending less than they would to build and also by delivering a very high quality solution that actually works well in production. As you can see by the current setup, it's not very good to say the least.

**Comment by alexandrui-telenav** [comment]
@jsniff thanks for the offer, you have a good point but we already trained a ML model that shows very good results and integrates well with the current processing pipeline.

**Comment by jsniff** [comment]
@alexandrui-telenav Alex, are you guys happy with your new technology? Let me know if you'd like to revisit discussions.

**Comment by alex9099** [comment]
It seems to still blur things that it shouldn't for example here, at the distance everything is blured https://openstreetcam.org/details/1469714/0/track-info
